azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_devops.go

```diff
@@ -0,0 +1,154 @@
+package serviceendpoint
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+const (
+	azdoPersonalAccessToken = "personal_access_token"
+)
+
+// ResourceServiceEndpointAzureDevOps schema and implementation for Azure DevOps service endpoint resource
+func ResourceServiceEndpointAzureDevOps() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointAzureDevOps, expandServiceEndpointAzureDevOps)
+	r.Schema["organization"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Azure DevOps organization name",
+	}
+	authPersonal := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			personalAccessToken: {
+				Type:         schema.TypeString,
+				Required:     true,
+				DefaultFunc:  schema.EnvDefaultFunc("AZDO_PERSONAL_ACCESS_TOKEN", nil),
+				Description:  "The Azure DevOps personal access token which should be used.",
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+		},
+	}
+	patHashKey, patHashSchema := tfhelper.GenerateSecreteMemoSchema(azdoPersonalAccessToken)
+	authPersonal.Schema[patHashKey] = patHashSchema
+	r.Schema["auth_personal"] = &schema.Schema{
+		Type:          schema.TypeSet,
+		Optional:      true,
```

Comment:
If I skip the  `auth_personal` configuration, after  execute `terraform apply`, I got error:
```log
Error: Error updating service endpoint in Azure DevOps: Unable to find service connection type 'azdoapi' using authentication scheme 'InstallationToken'.

  on main.tf line 5, in resource "azuredevops_serviceendpoint_devops" "serviceendpoint":
   5: resource "azuredevops_serviceendpoint_devops" "serviceendpoint" {


```

Reply:
added a unit test for `default` settings with empty `auth_personal` personal block